Extractor/Pushers/Writers/IdmWriter.cs

```diff
@@ -56,6 +58,25 @@ public IdmWriter(ILogger<IdmWriter> logger, CogniteDestinationWithIDM destinatio
             if (cleanConfig.Space == null) throw new ArgumentException("Attempted to initialize IDM writer without space ID");
             space = cleanConfig.Space;
             sources = new SourceSystemResource(destination.CogniteClient.Beta.DataModels);
+            var sourceId = cleanConfig.Source;
+            if (sourceId == null)
+            {
+                if (config.Source.EndpointUrl != null)
+                {
+                    sourceId = $"OPC_UA:{config.Source.EndpointUrl}";
+                }
+                else if (config.Source.NodeSetSource?.NodeSets?.Any() ?? false)
```

Comment:
Let's not forget to document this behavior clearly in the documentation as well as the example config when doing the release